firmware/boron-ins-fsm/src/stateMachine.cpp

```diff
@@ -126,10 +144,10 @@ void state0_idle() {
 
     Log.info("You are in state 0, idle: Door status, iAverage = 0x%02X, %f", checkDoor.doorStatus, checkINS.iAverage);
     // default timer to 0 when state doesn't have a timer
-    publishDebugMessage(0, checkDoor.doorStatus, checkINS.iAverage, 0);
+    publishDebugMessage(0, checkDoor.doorStatus, checkINS.iAverage, (millis() - timeSinceDoorClosed));
 
     // fix outputs and state exit conditions accordingly
-    if (((unsigned long)checkINS.iAverage > ins_threshold) && !isDoorOpen(checkDoor.doorStatus) && !isDoorStatusUnknown(checkDoor.doorStatus)) {
+    if (((unsigned long)checkINS.iAverage > ins_threshold) && !isDoorOpen(checkDoor.doorStatus) && !isDoorStatusUnknown(checkDoor.doorStatus) && millis() - timeSinceDoorClosed < state0_window_time) {
```

Comment:
While you're in here, there is an optimization that can be made. What do you think?

[hint: C++ (like many other languages) does short-circuit evaluation (https://www.geeksforgeeks.org/short-circuiting-in-c-and-linux/). ]

Reply:
I see, so I should move `millis() - timeSinceDoorClosed < state0_window_time` to the first check since it should be the most likely to fail?